lib/concurrent/edge/future.rb

```diff
@@ -348,7 +348,9 @@ def wait_until_complete(timeout)
           end
           break
         end
-        self
+        if completed?
+          return block ? block.call : self
+        end
```

Comment:
This is too heavy. `ns_wait_until` already returns true/false this method should have been doing the same. There should be `return` before `ns_wait_until` and no break. My bad.